ls1intum/Artemis
# `Programming exercises`: Fix test case type presentation for non Java exercises

### Checklist
#### General



#### Client

### Motivation and Context
Currently, we only support different test case types ('STRUCTURAL', 'BEHAVIORAL') for JAVA exercises. So for all other programming exercises, the test case type is 'DEFAULT'. Since there was no "DEFAULT" translation string, the UI broke there.

### Description
This PR adds some translation strings and fallbacks to display the test case type for all programming exercises.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 non-JAVA Programming Exercise e.g. Python 

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Python Exercise > Grading
4. Verify that the test case types display "Default"

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [programming-exercise-grading-tasks-table.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5650/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-grading-tasks-table.component.ts.html) | 90.76% | ✅ |
| [programming-exercise-task.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5650/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-task.service.ts.html) | 93.65% | ✅ |
| [programming-exercise-task.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5650/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-task.ts.html) | 100% | ✅|


### Review Progress






#### Code Review
#### Manual Tests

### Screenshots
Before:
![image](https://github.com/ls1intum/Artemis/assets/78978542/3a9b9926-85f2-4268-8a6d-7c8c0f82041c)

After:
![image](https://github.com/ls1intum/Artemis/assets/78978542/3dddc3ad-20a8-4668-a348-8600d19b98a2)
